### PR TITLE
Newrelic php 8.2 and 8.3

### PR DESCRIPTION
**Pull request by DracoBlue**
---
name: 🎉 New Feature
about: Add newrelic.so also for php8.2 and 8.3
labels: enhancement

---

### Description

This PR adds newrelic for php 8.2 and php 8.3 like I added it for php 8.1 2 years ago at https://github.com/shivammathur/homebrew-extensions/pull/2433 :)

### CODE REVIEW

File: Formula/newrelic@8.2.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+require File.expand_path("../Abstract/abstract-php-extension", __dir__)
+
+# Class for Newrelic Extension
+class NewrelicAT82 < AbstractPhpExtension
+  init
+  desc "Newrelic PHP extension"
+  homepage "https://github.com/newrelic/newrelic-php-agent"
+  url "https://github.com/newrelic/newrelic-php-agent/archive/refs/tags/v10.15.0.4.tar.gz"
+  version "v10.15.0.4"
+  sha256 "6c27106e8fcd428b1f018fdf7c43513e2a4eb490"
```
Comment (shivammathur) [inline Formula/newrelic@8.2.rb:13]:
```suggestion
  sha256 "8ffb3d377ec8632a5a873a1a53614b73027c8a3fbebba29c3aba956635578e01"
```

File: Formula/newrelic@8.3.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+require File.expand_path("../Abstract/abstract-php-extension", __dir__)
+
+# Class for Newrelic Extension
+class NewrelicAT83 < AbstractPhpExtension
+  init
+  desc "Newrelic PHP extension"
+  homepage "https://github.com/newrelic/newrelic-php-agent"
+  url "https://github.com/newrelic/newrelic-php-agent/archive/refs/tags/v10.15.0.4.tar.gz"
+  version "v10.15.0.4"
+  sha256 "6c27106e8fcd428b1f018fdf7c43513e2a4eb490"
```
Comment (shivammathur) [inline Formula/newrelic@8.3.rb:13]:
```suggestion
  sha256 "8ffb3d377ec8632a5a873a1a53614b73027c8a3fbebba29c3aba956635578e01"
```

**Review by shivammathur — CHANGES_REQUESTED** [review]

**Comment by DracoBlue** [comment]
I will need to double check if the patches are still missing. I will remove the draft status afterwards.

File: Formula/newrelic@8.2.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      newrelic.daemon.port="/tmp/.newrelic82.sock"
+      newrelic.logfile="/var/log/newrelic_php_agent.log"
+      newrelic.daemon.logfile="/var/log/newrelic_daemon.log"
+    EOS
+  rescue error
+    raise error
+  end
+
+  def install
+    inreplace "agent/php_autorum.c", "return (char*)emalloc(len);", "return (char*)emalloc((unsigned)len);"
```
Comment (DracoBlue) [inline Formula/newrelic@8.2.rb:46]:
```suggestion
    inreplace "agent/config.m4", "-l:libprotobuf-c.a", "-lprotobuf-c"
```

File: Formula/newrelic@8.3.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      newrelic.daemon.port="/tmp/.newrelic83.sock"
+      newrelic.logfile="/var/log/newrelic_php_agent.log"
+      newrelic.daemon.logfile="/var/log/newrelic_daemon.log"
+    EOS
+  rescue error
+    raise error
+  end
+
+  def install
+    inreplace "agent/php_autorum.c", "return (char*)emalloc(len);", "return (char*)emalloc((unsigned)len);"
```
Comment (DracoBlue) [inline Formula/newrelic@8.3.rb:46]:
```suggestion
    inreplace "agent/config.m4", "-l:libprotobuf-c.a", "-lprotobuf-c"
```

**Comment by shivammathur** [comment]
:beers: @BrewTestBot has [triggered a merge](https://github.com/shivammathur/homebrew-extensions/actions/runs/10178451463).

**Pull request merged**